core/src/main/scala/org/apache/spark/eventhubs/EventHubsConf.scala

```diff
@@ -389,14 +389,15 @@ object EventHubsConf extends Logging {
    * @return trimmed [[EventHubsConf]]
    */
   private[spark] def trim(ehConf: EventHubsConf): EventHubsConf = {
+    // These are the options needed by Spark executors
+    val include = Seq("eventhubs.connectionString",
+                      "eventhubs.consumerGroup",
+                      "eventhubs.receiverTimeout",
+                      "eventhubs.operationTimeout",
+                      "useSimulatedClient")
+
+    ehConf.settings.asScala.filter(tuple => include.contains(tuple._1))
```

Comment:
u intend to return the filtered list right? shouldn't this be the last line of this method as opposed to `ehConf` ?

```
ehConf.settings.asScala.filter(tuple => include.contains(tuple._1))
```

Reply:
No, we're returning the `EventHubsConf` which holds the internal list. So, we filter the list, and then return the ehConf which wraps it.

Reply:
do you mean `filter` modifies the original collection - i doubt it!!
It should return a new settings map with the filtered collection - after running your lamda.

if the trim method is called infrequently & not called in hot path - something like - for every batch - having it to return a copy of settings should be fine too.

Reply:
Ah shoot, `retain` modifies the original collection. I'll update it...